test/plugins/adonis.spec.js

```diff
@@ -0,0 +1,327 @@
+          })
+        
+        agent.use(traces => {
+          const spans = traces[0]
+
+          expect(spans[0]).to.have.property('service', 'test')
+
+          expect(spans[1]).to.have.property('name', 'GET')
+          expect(spans[1].meta).to.have.property('component', 'adonis')
+          expect(spans[1].meta).to.have.property('span.kind', 'server')
+          expect(spans[1].meta).to.have.property('http.url', `http://localhost:${port}/api/user`)
+          expect(spans[1].meta).to.have.property('http.method', 'GET')
+          expect(spans[1].meta).to.have.property('http.status_code', '500')
+          expect(spans[1].meta).to.have.property('error', 'true')
+          }).then(done)
+          .catch(done)
+
+        this.server.listen('localhost', port, () => {
+        axios
+          .get(`http://localhost:${port}/api/user`)
+        })
+      })
+    })
+
+    withVersions(plugin, '@adonisjs/middleware-base', version => {
+      before(() => {
+        Context.getter('request', function () {
+          return new Request(this.req, this.res, new Config())
+        }, true)
+
+        Context.getter('response', function () {
+          const config = new Config()
+          config.set('app.http.jsonpCallback', 'callback')
+          return new Response(this.request, config)
+        }, true)
+
+        setupResolver()
+        ioc.autoload(path.join(__dirname, 'app'), 'App')
+
+        return agent.load(plugin, 'adonis').then(() => {})
+      })
+
+      after(() => {
+        return agent.close()
+      })
+
+      beforeEach(() => {
+        tracer = require(reqPath)
+        this.exception = Exception
+        this.server = new Server(Context, Route, new Logger(), this.exception)
+        this.server.bindExceptionHandler()
+
+        ioc.fake('Adonis/Exceptions/BaseExceptionHandler', () => {
+          return new BaseExceptionHandler()
+        })
+
+        return getPort().then(newPort => {
+          port = newPort
+        })
+      })
+
+      afterEach(() => {
+        RouteStore.clear()
+        ioc.restore()
+        this.exception.clear()
+        return this.server.close()
+      })
+
+      it('should do automatic instrumentation on app middleware', done => {
+        class AppMiddleware {
+          async handle (ctx, next, [id]) {
+            ctx.ids = ctx.ids || []
+            ctx.ids.push(id)
+            await next()
+          }
+        }
+    
+        ioc.fake('Middleware/AppMiddleware', function () {
+          return new AppMiddleware()
+        })
+    
+        Route.get('/api/user', async function ({ ids }) {
+          return ids
+        })
+          .middleware('app:1')
+    
+        this.server.registerNamed({
+          'app': 'Middleware/AppMiddleware'
+        })
+    
+      agent.use(traces => {
+        const spans = traces[0]
+
+        expect(spans[0]).to.have.property('service', 'test')
+        expect(spans[0]).to.have.property('name', '<anonymous>')
```

Comment:
Is it possible to have a better default like `adonis.middleware` in cases where anonymous would be used?  It's not too informative otherwise.